### PR TITLE
feat(model-ad): support data filtering in Gene Expression CT (MG-598, MG-679)

### PR DESCRIPTION
**Pull request by rrchai**
## Description

This PR implements data filtering functionality for the Gene Expression comparison tool, allowing users to filter results by biological domains, model type, and mouse model name. 

Additional fix:
The gene expression endpoint was returning 500 errors `Field 'collation' is currently not supported
` in AWS dev. The code used `MongoDB's Collation.of("en").strength(Collation.ComparisonLevel.secondary())` in `main` branch for case-insensitive string sorting, which works in local MongoDB but is not supported by AWS DocumentDB. To fix this, the unsupported collation logic was replaced with `$toLower` aggregation operator (only apply to string-type fields), which is supported by both MongoDB and DocumentDB. cc @sagely1

## Related Issue

[MG-598](https://sagebionetworks.jira.com/browse/MG-598)
[MG-679](https://sagebionetworks.jira.com/browse/MG-679)

## Changelog

- Add data filtering query parameters (`biodomains`, `modelType`, `name`) to Gene Expression search API
- Implement custom MongoDB repository with aggregation pipeline for dynamic query construction and computed field sorting
- Add query parameter validation to reject unsupported or incorrectly-cased parameters
- Refactor service layer to delegate all query logic to repository layer
- Update OpenAPI schema with camelCase naming convention for consistency with Model Overview
- Add comprehensive test coverage for filtering combinations and edge cases
- Fix AWS DocumentDB compatibility by replacing MongoDB collation with `$toLower` aggregation for case-insensitive sorting in Gene Expression queries

### Build Docker Images

```
model-ad-build-images
```

### Start Full Stack

```
nx serve-detach model-ad-apex
```

#### Scenario A: Single Field Filter

```
http://localhost:8000/api/v1/comparison-tools/gene-expression?categories=RNA - DIFFERENTIAL EXPRESSION&categories=Tissue - Hemibrain&categories=Sex - Females&itemFilterType=exclude&biodomains=Apoptosis&biodomains=Cell Cycle&sortFields=gene_symbol&sortOrders=1
```
```console
{
  "geneExpressions": [
    {
      "gene_symbol": "1600012H06Rik",
      "name": "5xFAD (IU/Jax/Pitt)",
      "biodomains": ["Apoptosis"],
      "model_type": "Familial AD"
    },
    ...
  ],
  "page": {
    "totalElements": 13139
  }
}
```

#### Scenario B: Multi-Field Filter

```
http://localhost:8000/api/v1/comparison-tools/gene-expression?categories=RNA - DIFFERENTIAL EXPRESSION&categories=Tissue - Hemibrain&categories=Sex - Females&itemFilterType=exclude&biodomains=Apoptosis&modelType=Familial AD&sortFields=gene_symbol&sortOrders=1
```
```console
{
  "geneExpressions": [
    {
      "gene_symbol": "1600012H06Rik",
      "name": "5xFAD (IU/Jax/Pitt)",
      "biodomains": ["Apoptosis"],
      "model_type": "Familial AD"
    },
    ...
  ],
  "page": {
    "totalElements": 1948
  }
}
```

#### Scenario C: Invalid Field

```
http://localhost:8000/api/v1/comparison-tools/gene-expression?categories=RNA - DIFFERENTIAL EXPRESSION&categories=Tissue - Hemibrain&categories=Sex - Females&fake_gene_field=Value&sortFields=gene_symbol&sortOrders=1
```
```console
{
  "title": "Bad Request",
  "status": 400,
  "detail": "Unknown query parameter: fake_gene_field"
}
```

[MG-598]: https://sagebionetworks.jira.com/browse/MG-598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[MG-679]: https://sagebionetworks.jira.com/browse/MG-679?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/model-ad/api-next/src/main/java/org/sagebionetworks/model/ad/api/next/api/GeneExpressionApiDelegateImpl.java [hunk 1]
```diff
@@ -98,4 +120,17 @@ private String extractValueAfterPrefix(String value, String prefix, String field
 
     return extracted;
   }
+
+  private void validateQueryParameters() {
+    HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder
+      .getRequestAttributes())
```
[the remark this answers is not in the document]
Comment (rrchai) [reply apps/model-ad/api-next/src/main/java/org/sagebionetworks/model/ad/api/next/api/GeneExpressionApiDelegateImpl.java]: Good point, I will adopt this suggestion for the model overview as well to stay consistent
